src/kcas.ml

```diff
@@ -189,11 +190,16 @@ let cas loc before state =
 module Loc = struct
   type 'a t = 'a loc
 
-  let make after =
-    { state = Atomic.make @@ new_state after; id = Id.get_unique () }
+  let make ?(mode = Mode.obstruction_free) after =
+    {
+      state = Atomic.make @@ new_state after;
+      id = Id.get_unique ();
+      mode = mode;
```

Comment:
BTW, OCaml allows one to write this as just `mode`.  It is called [record field punning](https://dev.realworldocaml.org/records.html).

Reply:
ok.